standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -193,10 +194,11 @@ public boolean isDeadlock(SQLException e) {
 
   /**
    * Is the given exception a table not found exception
-   * @param e Exception
+   * @param ex Exception
    * @return
    */
-  public boolean isTableNotExistsError(SQLException e) {
+  public boolean isTableNotExistsError(Throwable ex) {
```

Comment:
why is this change?

Reply:
The logic slightly changed: Now the method accepts any kind of exceptions and finds an `SQLException` within the exception chain. If no `SQLException` found an `IllegalArgumentException` is thrown. This makes the caller code slightly less complex, as there is no need to find the `SQLException` if it's not the top one.